lesson 1/README.md

```diff
@@ -2,6 +2,93 @@
 
 ##  Github
 
+### Ideas behind a perfect repo
+
+#### 1. What should the master branch contain?
+Working and well-tested code that has been reviewed by peers.
+
+#### 2. Non-master branches?
+    ..1. Each of these branch should mostly have one (maybe two if you are pair-programming a feature) person working on it. 
+    ..2. Each branch should add a significant working change to the master. Usually each branch adds a working feature to the master
```

Comment:
Yes but it is pretty rare. I would say "Branches are ... but there might be situations when you need to make a small change that eveyone needs to see"